server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileSetDefaultAction.java

```diff
@@ -0,0 +1,106 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.qualityprofile.ws;
+
+import com.google.common.base.Preconditions;
+import org.sonar.api.resources.Languages;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.server.ws.WebService.NewAction;
+import org.sonar.core.permission.GlobalPermissions;
+import org.sonar.server.exceptions.NotFoundException;
+import org.sonar.server.qualityprofile.QProfile;
+import org.sonar.server.qualityprofile.QProfileFactory;
+import org.sonar.server.qualityprofile.QProfileLookup;
+import org.sonar.server.user.UserSession;
+
+import static org.apache.commons.lang.StringUtils.isEmpty;
+
+public class QProfileSetDefaultAction implements BaseQProfileWsAction {
+
+  private static final String PARAM_LANGUAGE = "language";
+  private static final String PARAM_PROFILE_NAME = "profileName";
+  private static final String PARAM_PROFILE_KEY = "profileKey";
+
+  private final Languages languages;
+
+  private final QProfileLookup profileLookup;
+
+  private final QProfileFactory profileFactory;
+
+  public QProfileSetDefaultAction(Languages languages, QProfileLookup profileLookup, QProfileFactory profileFactory) {
+    this.languages = languages;
+    this.profileLookup = profileLookup;
+    this.profileFactory = profileFactory;
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    UserSession.get().checkLoggedIn().checkGlobalPermission(GlobalPermissions.QUALITY_PROFILE_ADMIN);
+
+    String language = request.param(PARAM_LANGUAGE);
+    String profileName = request.param(PARAM_PROFILE_NAME);
+    String profileKey = request.param(PARAM_PROFILE_KEY);
+
+    Preconditions.checkArgument(
+        (!isEmpty(language) && !isEmpty(profileName)) ^ !isEmpty(profileKey), "Either profileKey or profileName + language must be set");
+
+    if(profileKey == null) {
+      profileKey = getProfileKeyFromLanguageAndName(language, profileName);
+    }
+
+    profileFactory.setDefault(profileKey);
+
+    response.noContent();
+  }
+
+  private String getProfileKeyFromLanguageAndName(String language, String profileName) {
+    QProfile profile = profileLookup.profile(profileName, language);
+    if (profile == null) {
+      throw new NotFoundException(String.format("Unable to find a profile for language '%s' with name '%s'", language, profileName));
+    }
+    return profile.key();
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
```

Comment:
It's a personal point of view, but I would put this in the top of the class, it allow to first see the description and the parameter before looking the implementation.